src/bin/pg_autoctl/cli_get_set_properties.c

```diff
@@ -0,0 +1,357 @@
+	}
+
+	if (!monitor_init(&(keeper.monitor), keeper.config.monitor_pguri))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	name = argv[0];
+	value = argv[1];
+
+	if (strcmp(name, "candidate-priority") == 0)
+	{
+		int candidatePriority = strtol(value, NULL, 10);
+		if (errno == EINVAL || candidatePriority < 0 || candidatePriority > 100)
+		{
+			log_error("candidate-priority value %s is not valid."
+					  " Valid values are integers from 0 to 100. ", value);
+			exit(EXIT_CODE_BAD_ARGS);
+		}
+
+		if (!monitor_set_node_candidate_priority(&(keeper.monitor),
+					keeper.state.current_node_id, keeper.config.nodename,
+					keeper.config.pgSetup.pgport, candidatePriority))
+		{
+			log_error("unable to set candidate-priority.");
+		}
+
+	}
+	else if (strcmp(name, "replication-quorum") == 0)
+	{
+		bool replicationQuorum = false;
+
+		if (!streq(value, "true") && !streq(value, "false")  )
+		{
+			log_error("replication-quorum value %s is not valid."
+					  " Valid values are true or false.", value);
+			exit(EXIT_CODE_BAD_ARGS);
+		}
+
+		replicationQuorum = streq(value, "true");
+		if (!monitor_set_node_replication_quorum(&(keeper.monitor),
+					keeper.state.current_node_id, keeper.config.nodename,
+					keeper.config.pgSetup.pgport, replicationQuorum))
+		{
+			log_error("unable to set replication-quorum.");
+		}
+	}
+	else
+	{
+		log_error("Unknown node property %s", name);
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+}
+
+
+/*
+ * cli_set_formation_property sets a formation property on the monitor
+ * for a formation the current keeper node belongs to.
+ */
+static void
+cli_set_formation_property(int argc, char **argv)
+{
+	KeeperConfig config = keeperOptions;
+	Monitor monitor = { 0 };
+	bool missing_pgdata_is_ok = false;
+	bool pg_is_not_running_is_ok = true;
+	char *name = NULL;
+	char *value = NULL;
+	int numberSyncStandbys = -1;
+
+	if (argc != 2)
+	{
+		log_error("Expected 2 but found %d arguments", argc);
+		commandline_help(stderr);
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	name = argv[0];
+	value = argv[1];
+
+	if (strcmp(name, "number-sync-standbys") == 0)
+	{
+		numberSyncStandbys = strtol(value, NULL, 10);
+		if (errno == EINVAL || numberSyncStandbys < 0)
+		{
+			log_error("number-sync-standbys value %s is not valid."
+					  " Expected a non-negative integer value. ", value);
+			exit(EXIT_CODE_BAD_ARGS);
+		}
+
+	}
+	else
+	{
+		log_error("Unknown formation property %s", name);
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!keeper_config_read_file(&config,
+								 missing_pgdata_is_ok,
+								 pg_is_not_running_is_ok))
+	{
+		/* errors have already been logged. */
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	if (!monitor_init_from_pgsetup(&monitor, &config.pgSetup))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!monitor_set_formation_number_sync_standbys(&monitor, config.formation, numberSyncStandbys))
+	{
+		log_error("Unable to set number-sync-standbys.");
```

Comment:
Review error message, include more information, ensure non-zero EXIT_CODE_... in that case, and print out the new value when we could set it.